### PR TITLE
[9.x] Add `isWithParameters` method and an alias named `isWith`

### PR DESCRIPTION
**Pull request by Namvarii**
With using Route::isWith() we can extend Route::is() functionality to check named routes with parameters.

Usage :
```
Route::isWith(
    'without.parameter', 
    ['parameters.removed'], 
    ['with.parameter', 'john'],
    ['with.parameter', ['name' => 'john']]
) {
    //
}
```

### CODE REVIEW

File: src/Illuminate/Routing/Router.php [hunk 1]
```diff
@@ -1199,6 +1199,40 @@ public function currentRouteNamed(...$patterns)
...
+     */
+    public function isWithParameters(...$patterns)
+    {
+        foreach ($patterns as $pattern) {
+            if (is_string($pattern)) {
+                if ($this->currentRouteNamed([$pattern])) {
+                    return true;
+                }
+            } else {
+                if (url()->current() == route($pattern[0], $pattern[1] ?? [])) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Router.php:1227]: Can't make call to foundation helper function here.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Routing/Router.php (hunk 1 above)
Comment (namvarii) [reply src/Illuminate/Routing/Router.php:1227]: Thanks, We need full URI of site, what is your suggestion to get that without foundation helper?

Comment (namvarii) [reply src/Illuminate/Routing/Router.php]: Now using `$this->getCurrentRequest()->url()` rather than `url()->current()`

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**